parasolr/tests/test_query.py

```diff
@@ -117,6 +174,29 @@ def test_filter(self):
         assert 'date:1500' in filtered_qs.filter_qs
         assert 'name:he*' in filtered_qs.filter_qs
 
+    def test_facet(self):
+        mocksolr = Mock(spec=SolrClient)
+        sqs = SolrQuerySet(mocksolr)
+        # facet a search
+        facet_list = ['person_type', 'item_type']
+        faceted_qs = sqs.facet(*facet_list)
+        # faceting should be set on
+        assert faceted_qs.facet_opts['facet']
+        assert faceted_qs.facet_opts['facet.field'] == facet_list
+        # a call to another method should leave facet options as is
+        faceted_qs = faceted_qs.filter(foo='bar')
+        assert faceted_qs.facet_opts['facet']
+        assert faceted_qs.facet_opts['facet.field'] == facet_list
+        # subsequents calls to facet should simply reset list
+        facet_list = ['foobars']
+        faceted_qs = faceted_qs.facet(*facet_list)
+        assert faceted_qs.facet_opts['facet']
+        assert faceted_qs.facet_opts['facet.field'] == facet_list
+        # kwargs should simply be set in facet opts
+        faceted_qs = faceted_qs.facet(*facet_list, **{'facet.sort': 'count'})
```

Comment:
Ah! Seeing this now I see I was unclear. I want to be able to say something like `sqs.facet('member', sort='count')`. If you look at the highlighting you'll see it works that way - the prefix is added in `query_opts`